### PR TITLE
K8SPXC-604: add owner references to webhook

### PR DESCRIPTION
**Pull request by qJkee**
[K8SPXC-604](https://jira.percona.com/browse/K8SPXC-604) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

Set operator deployment as webhook owner to make it automatically deleted when operator is removed

### CODE REVIEW

**Review by delgod — APPROVED** [review]

File: pkg/webhook/hook.go [hunk 1]
```diff
@@ -31,33 +33,54 @@ var hookPath = "/validate-percona-xtradbcluster"
...
+	}
+
+	ref, err := k8s.OwnerRef(operatorDeployment, h.scheme)
+	if err != nil {
+		return errors.Wrap(err, "failed to get deployment owner ref")
+	}
+
+	err = h.createService(ref)
+	if err != nil {
+		return errors.Wrap(err, "Can't create service")
```
Comment (blez) [inline pkg/webhook/hook.go:63]: should be lowercase

File: pkg/webhook/hook.go [hunk 2]
```diff
@@ -118,7 +143,12 @@ func (h *hook) createWebhook() error {
 			},
 		},
 	}
+
 	err := h.cl.Create(context.TODO(), hook)
+	if k8serrors.IsForbidden(err) {
+		return nil
```
Comment (blez) [inline pkg/webhook/hook.go:149]: Should we log it?

Comment (qJkee) [reply pkg/webhook/hook.go:149]: no, it's confusing message)
Since our webhook is fully optional

File: pkg/webhook/hook.go [hunk 3]
```diff
@@ -129,6 +159,7 @@ func (h *hook) createWebhook() error {
 		}
 
 		hook.Webhooks[0].ClientConfig.CABundle = h.caBunlde
+		hook.ObjectMeta.OwnerReferences = []metav1.OwnerReference{ownerRef}
```
Comment (blez) [inline pkg/webhook/hook.go:162]: Do this lines needed? It's already in hook object

File: pkg/webhook/hook.go [hunk 4]
```diff
@@ -77,19 +100,21 @@ func (h *hook) createService() error {
 			}
 
 			service.Spec.Selector = opPod.Labels
+			service.ObjectMeta.OwnerReferences = []metav1.OwnerReference{ownerRef}
```
Comment (blez) [inline pkg/webhook/hook.go:103]: Do this lines needed? It's already in service object

**Review by blez — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| big-data | passed |
| haproxy | passed |
| storage | passed |
| scaling | passed |
| upgrade-haproxy | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| self-healing | failed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| limits | failed |
| demand-backup | passed |
| smart-update | passed |
| monitoring-2-0 | passed |
| self-healing-advanced | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| operator-self-healing | passed |
| pitr | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | passed |
| storage | passed |
| self-healing | failed |
| upgrade-haproxy | passed |
| big-data | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| scheduled-backup | failed |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| big-data | passed |
| one-pod | passed |
| scaling | passed |
| upgrade-haproxy | passed |
| auto-tuning | passed |
| scaling-proxysql | passed |
| proxysql-sidecar-res-limits | passed |
| security-context | passed |
| recreate | passed |
| users | passed |
| restore-to-encrypted-cluster | passed |
| tls-issue-self | passed |
| demand-backup | passed |
| tls-issue-cert-manager | passed |
| pitr | passed |
| tls-issue-cert-manager-ref | passed |
| demand-backup-encrypted-with-tls | passed |
| validation-hook | passed |
| scheduled-backup | passed |
| storage | passed |
| self-healing | failed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |

**Pull request merged**